src/mender/statemachine/statemachine.py

```diff
@@ -115,9 +123,13 @@ def run(self, force_bootstrap=False):
 
 class Authorize(State):
     def run(self, context):
+        if not context.retry_timer.is_it_time():
+            log.debug("Not Authorizing (not time yet)")
+            time.sleep(1)
+            return
```

Comment:
Since this function has another return value, I think it make sense to explicitly point out what this returns:

```suggestion
            return None
```

(The function returns either `None` or the result of `authorize.request()`)

Reply:
sure. thanks for jumping in. changed in remaining 2 places as well.